src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -138,7 +138,7 @@ TelemetryClient telemetryClient
             }
         }
 
-        return _storeResponseBuilder.BuildResponse(requiredStudyInstanceUid);
+        return _storeResponseBuilder.BuildResponse(requiredStudyInstanceUid, _enableDropInvalidDicomJsonMetadata);
```

Comment:
Instead of passing the feature flag status, we can determine the leniency applied or not here and depends on that the response builder can set the status